src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -827,3 +1119,108 @@ def __removed_older_complete_status_files(self, status_folder):
+            if substatus_name['name'] == substatus_list_name:
+                return substatus_index
+        return None
+
+    def __recompose_truncated_status_file(self, truncated_status_file, truncated_package_list, count_total_errors, truncated_substatus_msg, substatus_index):
+        """ Recompose final truncated status file version """
+        truncated_detail_list = []
+        code, errors_details = self.__get_current_complete_status_errors(substatus_msg=truncated_substatus_msg)
+
+        # Check for existing errors before recompose
+        if code != Constants.PatchOperationTopLevelErrorCode.ERROR:
+            truncated_status_file['status']['substatus'][substatus_index]['status'] = Constants.STATUS_WARNING.lower()      # Update substatus status to warning
+        else:
+            truncated_detail_list.extend(errors_details)
+
+        truncated_msg_errors = self.__recompose_substatus_msg_errors(truncated_detail_list, count_total_errors)    # Recompose substatus msg errors
+        truncated_substatus_msg = self.__update_substatus_msg(substatus_msg=truncated_substatus_msg, substatus_msg_patches=truncated_package_list, substatus_msg_errors=truncated_msg_errors)
+        truncated_status_file['status']['substatus'][substatus_index]['formattedMessage']['message'] = json.dumps(truncated_substatus_msg)
+
+        return truncated_status_file
+
+    def __recompose_substatus_msg_errors(self, truncation_detail_list, count_total_errors):
+        """ Recompose truncated substatus errors json """
+        error_msg = Constants.StatusTruncationConfig.TRUNCATION_WARNING_MESSAGE
+        truncated_error_detail = self.__set_error_detail(Constants.PatchOperationErrorCodes.TRUNCATION, error_msg)  # Reuse the errors object set up
+        self.__try_add_error(truncation_detail_list, truncated_error_detail)
+        truncated_errors_json = self.__set_errors_json(count_total_errors, truncation_detail_list, True)    # True for truncated
+
+        return truncated_errors_json
+
+    def __update_substatus_msg(self, substatus_msg, substatus_msg_patches, substatus_msg_errors=None):
+        substatus_msg['patches'] = substatus_msg_patches
+        if substatus_msg_errors:
+            substatus_msg['errors'] = substatus_msg_errors
+        return substatus_msg
+
+    def __get_current_complete_status_errors(self, substatus_msg):
+        """ Get the complete status file errors code and errors details """
+        return substatus_msg['errors']['code'], substatus_msg['errors']['details']
+
+    def __create_assessment_tombstone_list(self, packages_removed_from_assessment):
+        assessment_tombstone_map = {}
+        tombstone_record_list = []
+
+        # Map['classification', count]
+        for package in packages_removed_from_assessment:
+            classifications = package['classifications'][0]
+            assessment_tombstone_map[classifications] = assessment_tombstone_map.get(classifications, 0) + 1
```

Comment:
assessment_tombstone_map.get()

Wouldn't this throw an error on empty list?

Reply:
done. if that's the case, this code is unreachable:

line 1000
            if len(packages_removed_from_assessment) > 0:
                # Update current assessment # of removed packages
                self.__assessment_packages_removed = packages_removed_from_assessment